frameworks/kafka/docs/managing.md

```diff
@@ -28,3 +28,56 @@ Increase the `BROKER_COUNT` value via the DC/OS web interface as in any other co
 1.  Install the latest version of Kafka:
 
         $ dcos package install kafka -—options=options.json
+
+# Graceful Shutdown
+## Extend Kill Grace Period
+
+Increase the `brokers.kill_grace_period` value via the DC/OS CLI, i.e. to `60`
+seconds. This example assumes that the Kafka service instance is named `kafka`.
+
+During the reconfiguration, each of the Kafka broker tasks are restarted. During
+the shutdown portion of the task restart, the previous configuration value for
+`brokers.kill_grace_period` is in effect. Following the shutdown, each broker
+task is launched with the new effective configuration value. So, care should be
+taken to monitor the period of time that Kafka brokers are taking to cleanly
+shutdown. The relevant log entries are enumerated within Configure.
+
+Create an options file `kafka-options.json` with the following content:
+
+        {
+            "brokers": {
+                "kill_grace_period": 60
+            }
+        }
+
+Issue the following command:
+
+        dcos kafka --name=/kafka update --options=kafka-options.json
+
+## Restart a Broker with Grace
+
+When restarting a Kafka broker, when the Kill Grace Period has been configured
+to a period that sufficiently allows Kafka to cleanly shutdown, the broker's
+shutdown time will be longer, but the subsequent startup time will be much more
+rapid as the Byzantine reconciliation activities otherwise required for startup
```

Comment:
Byzantine in the sense that the local broker knows/believes that it is in error based on the file that indicates a graceful shutdown is in progress is still present, so the broker needs to perform multiple reconciliation actions w/ peers. For consistency w/ the rest of our docs, I didn't feel that this level of detail should be present here. But I did feel that it is consistent and easier to reason about if we provide at least a brief mention as to why the startup would be slower following an un-graceful shutdown.